hadoop/src/main/java/org/apache/iotdb/tsfile/hadoop/TSFInputSplit.java

```diff
@@ -141,47 +85,203 @@ public long getLength() throws IOException, InterruptedException {
     return this.hosts;
   }
 
+  public List<ChunkGroupInfo> getChunkGroupInfoList() {
+    return chunkGroupInfoList;
+  }
+
+  public void setChunkGroupInfoList(List<ChunkGroupInfo> chunkGroupInfoList) {
+    this.chunkGroupInfoList = chunkGroupInfoList;
+  }
+
   @Override
   public void write(DataOutput out) throws IOException {
     out.writeUTF(path.toString());
-    out.writeLong(start);
     out.writeLong(length);
     out.writeInt(hosts.length);
-    for (int i = 0; i < hosts.length; i++) {
-      String string = hosts[i];
+    for (String string : hosts) {
       out.writeUTF(string);
     }
-    out.writeInt(numOfDeviceRowGroup);
-    RowGroupBlockMetaData rowGroupBlockMetaData = new TsRowGroupBlockMetaData(
-        deviceRowGroupMetaDataList)
-        .convertToThrift();
-    ReadWriteThriftFormatUtils
-        .writeRowGroupBlockMetadata(rowGroupBlockMetaData, (OutputStream) out);
+    out.writeInt(chunkGroupInfoList.size());
+    ChunkGroupInfo.convertToThrift(chunkGroupInfoList, out);
   }
 
   @Override
   public void readFields(DataInput in) throws IOException {
     path = new Path(in.readUTF());
-    this.start = in.readLong();
-    this.length = in.readLong();
+    length = in.readLong();
     int len = in.readInt();
-    this.hosts = new String[len];
+    hosts = new String[len];
     for (int i = 0; i < len; i++) {
       hosts[i] = in.readUTF();
     }
-    this.numOfDeviceRowGroup = in.readInt();
-    TsRowGroupBlockMetaData tsRowGroupBlockMetaData = new TsRowGroupBlockMetaData();
-    tsRowGroupBlockMetaData
-        .convertToTSF(ReadWriteThriftFormatUtils.readRowGroupBlockMetaData((InputStream) in));
-    deviceRowGroupMetaDataList = tsRowGroupBlockMetaData.getRowGroups();
+    int numOfChunkGroupInfo = in.readInt();
+    chunkGroupInfoList = ChunkGroupInfo.readFromThrift(numOfChunkGroupInfo, in);
   }
 
   @Override
   public String toString() {
-    return "TSFInputSplit [path=" + path + ", numOfDeviceGroup=" + numOfDeviceRowGroup
-        + ", deviceRowGroupMetaDataList=" + deviceRowGroupMetaDataList + ", start=" + start
-        + ", length="
-        + length + ", hosts=" + Arrays.toString(hosts) + "]";
+    return "TSFInputSplit [path=" + path + ", chunkGroupInfoList=" + chunkGroupInfoList
+        + ", length=" + length + ", hosts=" + Arrays.toString(hosts) + "]";
+  }
+
+
+  /**
+   * Used to store the key info in chunkGroupMetadata
+   */
+  public static class ChunkGroupInfo {
+    /**
+     * Name of device.
+     */
+    private String deviceId;
+
+    /**
+     * Name Of all the sensors in the device.
+     */
+    private String[] measurementIds;
+
+    /**
+     * Byte offset of the corresponding data in the file
+     * Notice: include the chunk group header and marker.
```

Comment:
I don't think we have chunk group header.

Reply:
Aha, I just copy that from the ChunkGroupMetaData, I will fix it in both files.